ci/ray_ci/runner.py

```diff
@@ -42,66 +50,51 @@
 def main(
     targets: List[str],
     team: str,
-    concurrency: int,
-    shard: int,
+    workers: int,
+    worker_id: int,
+    parallelism_per_worker: int,
     size: str,
     run_flaky_tests: bool,
 ) -> None:
     if not bazel_workspace_dir:
         raise Exception("Please use `bazelisk run //ci/ray_ci`")
     os.chdir(bazel_workspace_dir)
 
+    docker_login()
     if run_flaky_tests:
         test_targets = _get_flaky_test_targets(team)
     else:
-        test_targets = _get_test_targets(targets, team, concurrency, shard, size)
+        test_targets = _get_test_targets(targets, team, workers, worker_id, size)
     if not test_targets:
         logging.info("No tests to run")
         return
-    logging.info(f"Running tests: {test_targets}")
-    _run_tests(test_targets)
-
-
-def _run_tests(test_targets: List[str]) -> None:
-    """
-    Run tests
-    """
-    bazel_options = (
-        subprocess.check_output([f"{bazel_workspace_dir}/ci/run/bazel_export_options"])
-        .decode("utf-8")
-        .split()
-    )
-    subprocess.check_call(
-        ["bazel", "test", "--config=ci"] + bazel_options + test_targets
+    logger.info(f"Running tests: {test_targets}")
+    return_code = run_tests(
+        test_targets,
+        ["DL=1 ./ci/env/install-dependencies.sh"],
+        parallelism_per_worker,
     )
+    sys.exit(return_code)
```

Comment:
this is not really a `return_code` of anything. if you really want to return a `return_code`, maybe return the first non-zero return code from the shareded runs or something.

Reply:
i just want to capture the passing/failing status; the caller (buildkite) does not use the actual exit code for any purpose, maybe i can rename it to status

Reply:
ok, then maybe return a bool or throw a special exception and try to catch it or something? returning a number feels that it can mean something.